tests/integration/S3_integration_test.py

```diff
@@ -0,0 +1,114 @@
+from moto import mock_aws
+import boto3
+import pytest
+import os
+import shutil
+import tarfile
+import urllib.request
+
+from main_data_processor import run
+from dane.config import cfg
+from io_util import untar_input_file, S3_OUTPUT_TYPES
+
+
+source_id = "carrier"
+basename_sourceid = "resource__" + source_id
+fn_tar_in = f"{basename_sourceid}.tar.gz"
+key_in = f"{cfg.INPUT.S3_FOLDER_IN_BUCKET}/{fn_tar_in}"
+tar_out = f"{source_id}/out__{source_id}.tar.gz"
+key_out = f"{cfg.OUTPUT.S3_FOLDER_IN_BUCKET}/{tar_out}"
+
+
+@pytest.fixture
+def aws_credentials():
+    """Create custom AWS setup: mocked AWS Credentials for moto."""
+    os.environ["AWS_ACCESS_KEY_ID"] = "testing"
+    os.environ["AWS_SECRET_ACCESS_KEY"] = "testing"
+    os.environ["AWS_SECURITY_TOKEN"] = "testing"
+    os.environ["AWS_SESSION_TOKEN"] = "testing"
+    os.environ["AWS_DEFAULT_REGION"] = "us-east-1"  # Other regions make stuff complex
+    os.environ["MOTO_S3_CUSTOM_ENDPOINTS"] = cfg.INPUT.S3_ENDPOINT_URL
+
+
+@pytest.fixture
+def aws(aws_credentials):
+    """Spin up local aws for testing"""
+    with mock_aws():
+        yield boto3.client("s3")
+
+
+@pytest.fixture
+def create_sample_input():
+    """
+    Add sample input for test to input bucket.
+    In this case, the sample input is downloaded from Openbeelden.
+    You can also add a file from the repository (e.g. from data/input-files/<example-input>)
+    """
+    fn = "inputfile.mp4"
+    urllib.request.urlretrieve(
+        "https://www.openbeelden.nl/files/06/28/628929.628917.MERAM_Copy__2.mp4", fn
+    )
+    with tarfile.open(fn_tar_in, "w:gz") as tar:
+        tar.add(fn)
+    yield
+    # after test: cleanup
+    os.remove(fn)
+    os.remove(fn_tar_in)
+
+
+@pytest.fixture
+def create_and_fill_buckets(aws, create_sample_input):
+    """Make sure input and output buckets exist, and add sample input"""
+    client = boto3.client("s3")
+    for bucket in [
+        cfg.INPUT.S3_BUCKET,
+        cfg.OUTPUT.S3_BUCKET,
+        cfg.INPUT.S3_BUCKET_MODEL,
+    ]:
+        client.create_bucket(Bucket=bucket)
+    client.upload_file(
+        Filename=fn_tar_in,
+        Bucket=cfg.INPUT.S3_BUCKET,
+        Key=f"{cfg.INPUT.S3_FOLDER_IN_BUCKET}/{key_in}",
```

Comment:
I think key_in already has the `cfg.INPUT.S3_FOLDER_IN_BUCKET` included, right?

Reply:
Yes. I think I copied this part of code from the example worker, but I will double check to see if it is the same indeed

Reply:
Huh, so apparently, without the S3_FOLDER_IN_BUCKET twice, it doesn't work @Veldhoen 

Working on finding out the reason

Reply:
So the issue was that `cfg.INPUT.S3_FOLDER_IN_BUCKET` was present not only here, but also lower as an argument to the `run` function of `main_data_processor`. I removed it from both places there and I will create a PR with the same changes in dane-example-worker because the same issue is present there too